.buildkite/hooks/pre-command

```diff
@@ -105,3 +105,17 @@ EOF
 <summary>Agent information from gobld</summary>
 EOF
 fi
+
+# Amazon Linux 2 has DNS resolution issues with resource-based hostnames in EC2
+# We have many functional tests that try to lookup and resolve the hostname of the local machine in a particular way
+# And they fail. This sets up dnsmasq locally and sets up a manual entry for the hostname.
+if [[ -f /etc/os-release ]] && grep -q '"Amazon Linux 2"' /etc/os-release; then
+  sudo yum install -y dnsmasq bind-utils
```

Comment:
Can we do the package install stuff in our CI image building? We just want to avoid as much shit that might transiently fail at build time as possible.

Reply:
Done and updated